### PR TITLE
#1 Resource path in CreateContainer name should be converted to lowercase

### PR DESCRIPTION
**Pull request by bvbeek**
According : https://docs.microsoft.com/en-us/rest/api/storageservices/create-container

The Create Container request may be constructed as follows. HTTPS is recommended. Your mycontainer value can only include lower-case characters. Replace myaccount with the name of your storage account:

### CODE REVIEW

File: samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al [hunk 1]
```diff
@@ -416,13 +442,16 @@ codeunit 50171 AzureBlobStorage
         end;
 
         // Add Authorization header
+        resourcePath := TypeHelper.UriEscapeDataString(resourcePath).Replace('%2F', '/');
```
Comment (esbenk) [inline samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al]: Why this? Do you have an example where it fails?

File: samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al [hunk 2]
```diff
@@ -416,13 +442,16 @@ codeunit 50171 AzureBlobStorage
...
+
         token := GetSasToken(methodVerb, ResourcePath, UtcDateTime, ContentLength, ContentType, CreateCanonicalizedParameters(parameters), xHeaderList);
         headers.Add('Authorization', token);
 
         request.Method := methodVerb;
         if parameters <> '' then
-            resourcePath := resourcePath + '?' + parameters;
-        request.SetRequestUri(ResourceUri + ResourcePath);
+            resourcePath := resourcePath + '?' + parameters.Replace('/', '%2F');
```
Comment (esbenk) [inline samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al]: It is probably just the value part of the parameters that should be encoded using UriEscapeDataString(). Do you have an example where this is needed?

File: samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al [hunk 3]
```diff
@@ -283,6 +285,30 @@ codeunit 50171 AzureBlobStorage
         CheckResponseCode(response);
     end;
 
+    procedure DeleteBlobs(ResourcePath: Text; Prefix: Text);
```
Comment (esbenk) [inline samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al]: Please add a comment describing the method. I will try to put comments on the other public methods.

File: samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al [hunk 4]
```diff
@@ -124,6 +124,8 @@ codeunit 50171 AzureBlobStorage
     begin
         CheckInitialized();
 
+        ResourcePath := LowerCase(ResourcePath);
```
[the remark this answers is not in the document]
Comment (bvbeek) [reply samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al]: Good point!

LowerCaseError: label 'The container value "%1" can only include lower-case characters.';
if ResourcePath <> LowerCase(ResourcePath) then
  error(LowerCaseError,ResourcePath)

File: samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al (hunk 1 above)
Comment (bvbeek) [reply samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al]: This is needed to support a prefix with folder and subfolder support: ListBlobs -Prefix container/MainFolder/SubFolder/Document-

File: samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al (hunk 2 above)
Comment (bvbeek) [reply samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al]: This is needed to support a prefix with folder and subfolder support: ListBlobs -Prefix container/MainFolder/SubFolder/Document-

File: samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al (hunk 3 above)
Comment (bvbeek) [reply samples/AzureStorage/AzureStorageLibrary/Blob/AzureBlobStorage.Codeunit.al]: I think you can better describe the comment in your context ;)

What I think is: 
The DeleteBlobs operation delete a list of the blobs in a specified container. The prefix parameter delete only blobs whose names begin with the specified prefix.

**Pull request merged**